src/main/java/com/exadel/discount/service/impl/JwtGenerationServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.exadel.discount.service.impl;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.exadel.discount.service.JwtGenerationService;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.stream.Collectors;
+
+@Service
+@Setter
+@Slf4j
+public class JwtGenerationServiceImpl implements JwtGenerationService {
+    private final String ROLES_CLAIM_NAME = "role";
+    private final String TOKEN_ENCRYPTION_KEY = "encryptionSecret";
+    private final String REFRESH_ROLE = "ROLE_REFRESH";
+
+    @Value("${jwt.expiration.seconds.access}")
+    private long ACCESS_TOKEN_EXPIRATION_TIME;
+    @Value("${jwt.expiration.seconds.refresh}")
+    private long REFRESH_TOKEN_EXPIRATION_TIME;
+
+    @Override
+    public String generateAccessToken(UserDetails userDetails) {
+        log.debug("preparing an access token creation data");
+        return buildToken(
+                userDetails.getUsername(),
+                ACCESS_TOKEN_EXPIRATION_TIME,
+                userDetails.getAuthorities()
+                        .stream()
+                        .map(GrantedAuthority::getAuthority)
+                        .collect(Collectors.joining(" ")));
```

Comment:
Why still concatenation???
https://github.com/SergeyKoval/2021_june_team4_be/pull/28#discussion_r661287328

Reply:
This was resolved in the commit "52c5529".